_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_vaapi.cpp

```diff
@@ -928,9 +929,11 @@ mfxStatus VAAPIEncoder::QueryCompBufferInfo(D3DDDIFORMAT type, mfxFrameAllocRequ
     }
     else if (type == D3DDDIFMT_INTELENCODE_MBSEGMENTMAP)
     {
+        // driver seg map is always in 16x16 blocks because of HW limitation
+        constexpr mfxU16 blockSize = 16;
         request.Info.FourCC = MFX_FOURCC_VP9_SEGMAP;
-        request.Info.Width  = AlignValue(frameWidth >> 5, 64);
-        request.Info.Height = frameHeight >> 5;//VDENC 32x32 block size ??
+        request.Info.Width  = AlignValue(frameWidth / blockSize, 64);
```

Comment:
Could you please specify where 64 alignment comes from? Is it a driver's demand?

Reply:
why bytes? just 64 aligned... in pixels